src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx

```diff
@@ -45,43 +52,57 @@ class IntegrationServerlessRow extends React.Component<Props> {
     );
   };
   toggleEnable = async () => {
+    const {serverlessFunction} = this.props;
     const action = this.enabled ? 'disable' : 'enable';
     const data = {
       action,
-      target: this.props.serverlessFunction.name,
+      target: serverlessFunction.name,
     };
     try {
       addLoadingMessage();
+      this.setState({submitting: true});
+      //optimistically update enable state
+      this.props.onUpdateFunction({enabled: !this.enabled});
```

Comment:
Is it ok to set the `ServerlessFunction` with just `enabled`? Should it update the existing data without overwriting?

Reply:
@mgaeta good question. I do a merge operation at the higher level so we keep any fields that we do not change here. Does that make sense?

Reply:
Yup 👍🏽